[update]投稿検索のバリエーションを追加

ryuzo111 opened the pull request:
【説明】
投稿検索の条件のバリエーションを追加しました。

【コマンド】
特になし

【期待する結果】
今まで→タイトルのみ検索
これから→タイトル、悩みのタイプ、悩みの対象者、興味のある保険、日付で検索可能

ryuzo111 committed:
[fix]プロフィールページのメールアドレスの表示削除＆家族加入保険をログイン者のみみれる仕様に変更

ryuzo111 committed:
Merge branch 'master' of https://github.com/ryuzo111/matching_insurance

Conflicts:
	resources/views/profile/detail.blade.php

ryuzo111 committed:
[add]検索機能のバリエーションを増やす

komo15 commented on app/Http/Controllers/PostController.php:
```diff
@@ -36,13 +36,18 @@ public function index()
+        // dd(in_array(1, $request->trouble_type));
+        // dd(in_array(1, $request->insurance_target));
+        $posts = $this->post->getSearchResults($param);
+
```
コメントアウト部分は消した方が良いと思います。

komo15 commented on app/Models/Post.php:
```diff
@@ -27,17 +28,61 @@ public function comments()
+        $query = Post::query();
+        if (!empty($param['word'])) {
+            $query->where('title', 'like', '%' . $param['word'] . '%');
+        }
```
タイトルのみの検索とわかりにくいので、「タイトル検索」とつけるか、投稿者や本文の文言も検索できるのが良いかなと思いました。

ryuzo111 replied:
タイトルの他に、投稿者と本文の文言も検索できるように変更しました！

komo15 commented on app/Models/Post.php:
```diff
@@ -27,17 +28,61 @@ public function comments()
+                $all_interested_posts = $all_interested_posts->merge($post);
+            }
+
+            $all_interested_posts = $all_interested_posts->pluck('post_id')->unique();
```
ここの処理、なるほど！という感じでした！
https://procir.slack.com/archives/C0267BHQX08/p1625699318008200
ここで私もterukinaさんに教えていただいた、変数の再代入をなるべくしないという点、
どこまで厳密にやるかは私もつかみかねていますが、頻出していたので気になりました。
少なくとも、pluckで取った時の変数は$all_interested_posts_idとかにできるのかな、と思いました。

komo15 commented on app/Models/Post.php:
```diff
@@ -27,17 +28,61 @@ public function comments()
+        }
+        if (!empty($param['end_time'])) {
+            $query->whereDate('created_at', '>=', $param['end_time']);
+        }
```
こちら、end_time以降になってしまっているため、25日までのものを調べたくて入力しても、25日以前のものが出てこない仕様になっています。
また、start_timeは検索時に入力が保持されるんですが、end_timeは消えてしまいます。

ryuzo111 committed:
[fix]検索フォームの修正、ページネーションの2ページ目の設定

テキストの検索フォームでは、修正前はタイトルのみの検索から、タイトルと内容とユーザー名で検索できるように修正しました。ページネーションの2ページ目は検索の内容を引き継がなかったのですが、引き継ぐように修正しました。

ryuzo111 committed:
[fix]search.blade.phpの文言を修正
